src/lib/support/ThreadOperationalDataset.h

```diff
@@ -270,88 +191,219 @@ class OperationalDataset
+     * @param[in]   aMeshLocalPrefix   The Thread mesh local prefix.
+     *
+     * @retval CHIP_NO_ERROR           Successfully set the Thread mesh local prefix.
+     * @retval CHIP_ERROR_NO_MEMORY    Insufficient memory in the dataset for setting Thread mesh local prefix.
      */
-    static bool IsValid(ByteSpan aData);
+    CHIP_ERROR SetMeshLocalPrefix(const uint8_t (&aMeshLocalPrefix)[kSizeMeshLocalPrefix]);
 
-    ByteSpan AsByteSpan(void) const { return ByteSpan(mData, mLength); }
+    /**
+     * This method sets Thread network name to the dataset.
+     *
+     * @param[in]   aNetworkName    The Thread network name.
+     *
+     * @retval CHIP_NO_ERROR           Successfully set the network name.
+     * @retval CHIP_ERROR_NO_MEMORY    Insufficient memory in the dataset for setting Thread network name.
+     */
+    CHIP_ERROR SetNetworkName(const char * aNetworkName);
+
+    /**
+     * This method sets Thread PAN ID to the dataset.
+     *
+     * @param[in]   aPanId  The Thread PAN ID.
+     *
+     * @retval CHIP_NO_ERROR           Successfully set the PAN ID.
+     * @retval CHIP_ERROR_NO_MEMORY    Insufficient memory in the dataset for setting Thread PAN ID.
+     */
+    CHIP_ERROR SetPanId(uint16_t aPanId);
+
+    /**
+     * This method sets Thread PSKc to the dataset.
+     *
+     * @param[in]   aPSKc   The Thread PSKc.
+     *
+     * @retval CHIP_NO_ERROR           Successfully set the PSKc.
+     * @retval CHIP_ERROR_NO_MEMORY    Insufficient memory in the dataset for setting Thread PSKc.
+     */
+    CHIP_ERROR SetPSKc(const uint8_t (&aPSKc)[kSizePSKc]);
+
+    /**
+     * This method unsets Thread PSKc to the dataset.
+     */
+    void UnsetPSKc();
+
+    /**
+     * This method sets the channel mask within the dataset.
+     *
+     * @retval CHIP_NO_ERROR on success.
+     * @retval CHIP_ERROR_NO_MEMORY if there is insufficient space within the dataset.
+     */
+    CHIP_ERROR SetChannelMask(ByteSpan aChannelMask);
+
+    /**
+     * This method sets the security policy within the dataset.
+     *
+     * @retval CHIP_NO_ERROR on success.
+     * @retval CHIP_ERROR_NO_MEMORY if there is insufficient space within the dataset.
+     */
+    CHIP_ERROR SetSecurityPolicy(uint32_t aSecurityPolicy);
+
+    /**
+     * This method sets the delay timer within the dataset.
+     *
+     * @retval CHIP_NO_ERROR on success.
+     * @retval CHIP_ERROR_NO_MEMORY if there is insufficient space within the dataset.
+     */
+    CHIP_ERROR SetDelayTimer(uint32_t aDelayMillis);
+
+    /**
+     * This method clears all data stored in the dataset.
+     */
+    void Clear() { mData = ByteSpan(); }
+
+    /**
+     * Returns a ByteSpan view of the current state of the dataset.
+     * The byte span is only valid as long the OperationalDataset object is alive and not modified.
+     */
+    ByteSpan AsByteSpan() const { return this->OperationalDatasetView::AsByteSpan(); }
```

Comment:
Why is this needed, if we publicly inherit from OperationalDatasetView?  I guess the documentation about invariants is different?

Reply:
Yes this is just so I had somewhere to put the comment, not sure if there's a more elegant way to do that.